Mathlib/CategoryTheory/Triangulated/Triangulated.lean

```diff
@@ -0,0 +1,158 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+
+! This file was ported from Lean 3 source module category_theory.triangulated.triangulated
+! leanprover-community/mathlib commit 19786714ebe478f40b503acb4705fb058ba47303
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Triangulated.Pretriangulated
+
+/-!
+# Triangulated Categories
+
+This file contains the definition of triangulated categories, which are
+pretriangulated categories which satisfy the octahedron axiom.
+
+-/
+
+
+noncomputable section
+
+namespace CategoryTheory
+
+open Limits Category Preadditive Pretriangulated
+
+open ZeroObject
+
+variable (C : Type _) [Category C] [Preadditive C] [HasZeroObject C] [HasShift C ℤ]
+  [∀ n : ℤ, Functor.Additive (shiftFunctor C n)] [Pretriangulated C]
+
+namespace Triangulated
+
+-- porting note: this structure was added because otherwise Lean4 would complain that
+-- the definition of `Octahedron` contains free variables
```

Comment:
This minimizes all the way down to
```
-- (kernel) declaration has free variables 'O.mk.injEq'
structure O (c : False = False) where
  m : Prop
```
which is ... surely a bug?

Reply:
```suggestion
-- porting note: this structure was added because otherwise Lean4 would complain that
-- the definition of `Octahedron` contains free variables
-- This issue has been minimized and raised on zulip at https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/free.20variables.20in.20mk.2EinjEq
```